src/gdbstub_impl/ext/exec_file.rs

```diff
@@ -19,18 +17,14 @@ impl<T: Target, C: Connection> GdbStubImpl<T, C> {
 
         let handler_status = match command {
             ExecFile::qXferExecFileRead(cmd) => {
-                let offset = <T::Arch as Arch>::Usize::from_be_bytes(cmd.offset)
-                    .ok_or(Error::TargetMismatch)?;
-                let length = <T::Arch as Arch>::Usize::from_be_bytes(cmd.length)
-                    .ok_or(Error::TargetMismatch)?;
                 let ret = ops
-                    .get_exec_file(cmd.pid, offset, length, cmd.buf)
+                    .get_exec_file(cmd.pid, cmd.offset, cmd.length, cmd.buf)
                     .handle_error()?;
-                if ret.is_empty() {
+                if ret == 0 {
                     res.write_str("l")?;
                 } else {
                     res.write_str("m")?;
-                    res.write_binary(ret)?;
+                    res.write_binary(&cmd.buf[..ret])?;
```

Comment:
A malformed target implementation might return a `ret` that is greater than `cmd.buf` length, thereby introducing a panic in this indexing code.

I'm not _entirely_ sure how we'd want to handle this case, but I tend to err on the side of "fail loud and fail fast" rather than "try to do the vaguely right thing". As such, I'd swap this out for something like:

```
// TODO: add more specific error variant?
res.write_binary(cmd.buf.get(..ret).ok_or(Error::PacketBufferOverflow)?)?;
```

where the TODO is a reminder to potentially add a more specific error variant to handle these kinds of situations.